src/systems/physics/Physics.cc

```diff
@@ -1801,31 +1806,39 @@ std::unordered_map<Entity, physics::FrameData3d> PhysicsPrivate::ChangedLinks(
 
 //////////////////////////////////////////////////
 void PhysicsPrivate::UpdateSim(EntityComponentManager &_ecm,
-    const std::unordered_map<Entity, physics::FrameData3d> &_linkFrameData)
+    const std::map<Entity, physics::FrameData3d> &_linkFrameData)
 {
   IGN_PROFILE("PhysicsPrivate::UpdateSim");
 
   IGN_PROFILE_BEGIN("Models");
 
-  _ecm.Each<components::Model, components::ModelCanonicalLink>(
-      [&](const Entity &_entity, components::Model *,
-          components::ModelCanonicalLink *_canonicalLink) -> bool
```

Comment:
Thanks for catching this! You are correct - in fact, we no longer need to `ModelCanonicalLink` component anymore. I removed the usage of this component in https://github.com/ignitionrobotics/ign-gazebo/pull/736/commits/76cfcf4b4c25919b28278ee4b7d25d217c13bf97.

Since we no longer use `ModelCanonicalLink`, would it be worth deprecating or removing it? I'm assuming we cannot remove this component from Edifice since that would break ABI? Can we deprecate it in Edifice, or would deprecation also need to wait for Fortress?

I'm also copying @azeey on this thread in case he has any input, since he was the one who introduced the `ModelCanonicalLink` component in #685.

Reply:
>  Can we deprecate it in Edifice, or would deprecation also need to wait for Fortress?

I think deprecation will need to be done in Fortress (assuming we no longer want to keep this component)

Reply:
Deprecating it sounds good to me. I think we should also deprecate the `CanonicalLink` component since now we can check if a link is a canonical link by checking for the existence of `ReferenceModels`. 

I know in the past we have said we should not add a new required component on an entity in a minor release. From what I can tell, if the component `ReferenceModels` does not exist on a link (possibly because the model it belongs to was created by a user plugin without using `SdfEntityCreator`), then these pose updates won't function properly.  Do you have thoughts on this @chapulina?

Reply:
> From what I can tell, if the component ReferenceModels does not exist on a link (possibly because the model it belongs to was created by a user plugin without using SdfEntityCreator), then these pose updates won't function properly

I believe that this is correct. Are there currently ways for models to be created without using the `SdfEntityCreator`?

Reply:
AFAIK, the `SdfEntityCreator` is a convenient way to create models. Plugins can create models by themselves. Perhaps it's time to stipulate that all models should be created using `SdfEntityCreator`.

Reply:
I have modified the implementation of how canonical link mappings are achieved, so we no longer need to deprecate `ModelCanonicalLink` or `CanonicalLink` since they're still being used: https://github.com/ignitionrobotics/ign-gazebo/pull/736#issuecomment-830184043

> Perhaps it's time to stipulate that all models should be created using `SdfEntityCreator`.

I've opened a PR related to this for Fortress, comments/thoughts are welcome: #787